processor/transformprocessor/internal/traces/traces.go

```diff
@@ -532,3 +548,25 @@ func setAttr(attrs pcommon.Map, mapKey string, val interface{}) {
 		// TODO(anuraaga): Support set of map type.
 	}
 }
+
+func getTraceState(traceState ptrace.TraceState, mapKey string) interface{} {
+	ts, err := trace.ParseTraceState(string(traceState))
+	if err == nil {
+		return ts.Get(mapKey)
+	}
+	return nil
+}
+
+func setTraceState(traceState ptrace.TraceState, doSetTraceState func(v ptrace.TraceState), mapKey string, val interface{}) {
+	ts, err := trace.ParseTraceState(string(traceState))
+	if err != nil {
+		return
+	}
+	if str, ok := val.(string); ok {
```

Comment:
Let's do this `val` check before doing `trace.ParseTraceState`.  That way if `val` isn't a string we didn't waste any time doing `trace.ParseTraceState`.